aiondemand/AIOD-rest-api
# Docs/connectors

## Change

Further centralizes the different documentation files across the repository into one more structured documentation site.
It should be easier to navigate, find, and present information with the mkdocs plugins and on-page hierarchies.
I generally updated and expanded a lot of information, but in some cases I preferred to withhold since it would be better to improve the docs over time or to have people more familiar with a particular service write/update the docs.


## How to Test

To build the documentation locally, after checkout out this branch: 

* Install mkdocs-material: `python -m pip install mkdocs-material`
* Run the documentation server: `python -m mkdocs serve`

The documentation is already available at https://aiondemand.github.io/AIOD-rest-api/ -- I will add a CI/CD script later to push new documentation as part of a commit to `develop`.
This will also provide live updates if you edit the files.
I suggest each reviewer look at "their own" part of the documentation first. If time permits, feedback on the general sections are also welcome.

 * @mrorro: There is authentication information in both "hosting" and "developer", this is largely copied from the original README.
 * @mrorro: There is also a (fairly minimal) contributing page, which I believe we discussed last WP call. Perhaps you want to have a look.
 * @AdrianAlcolea: I added a page on Elastic Search and Logstash, this is only under "developer". We might need also information under "hosting", but I think it is fairly self explanatory.
 * @Taniya-Das: Using the original READMEs as a base, I expanded and revised for both the "hosting" and "developer" sections. I believe you already looked at this in an earlier iteration. If you could skim through to see if there's something that we need to improve _now_, that would be useful. Otherwise, we'll improve further as we go.

You can provide feedback by:
 - Branching off this branch and then setting up a PR into this branch, or
 - Use the GitHub "suggest change" options, navigate to the file in the PR either comment like normal, or from the comment box click the little file icon (or use `<Ctrl/Mac>+g`) to directly create a diff that you can propose, or

Try to avoid committing directly to this branch, unless you are comfortable with rebasing (as other commits may also happen).

## Checklist
  - No unintended changes have been committed.
  - The changes in isolation seem reasonable.
  - Anything that may be odd or unintuitive is provided with a GitHub comment explaining it (but consider if this should not be a code comment or in the documentation instead).

## Related Issues
